neps/nep-0455.md

```diff
@@ -0,0 +1,191 @@
+---
+NEP: 455
+Title: Parameter Compute Costs
+Author: Andrei Kashin <andrei.kashin@near.org>, Jakob Meier <jakob@near.org>
+DiscussionsTo: https://github.com/nearprotocol/neps/pull/455
+Status: Draft
+Type: Protocol Track
+Category: Runtime
+Created: 26-Jan-2023
+---
+
+## Summary
+
+Introduce compute costs decoupled from gas costs for individual parameters to allow to safely limit the compute time it takes to process the chunk.
+
+## Motivation
+
+For NEAR blockchain stability, we need to ensure that blocks are produced regularly and in a timely manner.
+
+The chunk gas limit is used to ensure that the time it takes to validate a chunk is strictly bounded by limiting the total gas cost of operations included in the chunk.
+This process relies on accurate estimates of gas costs for individual operations.
+
+Underestimating these costs leads to *undercharging* which can increase the chunk validation time and slow down the chunk production.
+
+As a concrete example, during the incremental deployment of Flat Storage we will be going through intermediate stages where the storage operations will be undercharged.
+
+## Rationale
+
+Separating compute costs from gas costs will allow us to safely limit the compute usage for processing the chunk while still keeping the gas prices the same and thus not breaking existing contracts.
+After adding compute cost for target parameter we will be able to publicly discuss the ways to solve the undercharging (optimize implementation, smart contract or increasing the gas cost).
+
+This design is easy to implement and simple to reason about and provides a clear way to address existing undercharging issues.
+
+If we don't address the undercharging problems, we increase the risks that they will be exploited and also make it harder to deploy Flat Storage safely.
```

Comment:
Not sure if we have sufficiently explained how this helps deploy flat storage safely yet.

Reply:
lgtm